cineast-core/src/main/java/org/vitrivr/cineast/core/temporal/ScoredSegment.java

```diff
@@ -0,0 +1,102 @@
+package org.vitrivr.cineast.core.temporal;
+
+import org.jetbrains.annotations.NotNull;
+import org.vitrivr.cineast.core.data.StringDoublePair;
+import org.vitrivr.cineast.core.data.entities.MediaSegmentDescriptor;
+
+/**
+ * Class to store the information of a scored segment.
+ *
+ * <p>Stores relevant information and score of a segment of a container</p>
+ * <p>Utilized in all instances of the {@link AbstractTemporalScoringAlgorithm} to add up all scores of a certain segment scored in a certain container</p>
+ * <p>Provides functions to compare two scored segments with each other according to their container id and segment id</p>
+ * <p>Each scored segment is the result of adding up all scores of all results from one SimilarityQuery concerning one segment Id</p>
+ */
+public class ScoredSegment implements Comparable<ScoredSegment> {
```

Comment:
Why isn't this at all related to the MediaSegmentDescriptor, even though it has all the same fields (as well as some additional ones)?

Reply:
You are right, however, I was not sure how inheritance would translate when serialising an instance of this class into JSON. Could you give me an estimate that it wouldn't break anything to replace the common code by extending MediaSegmentDescriptior?

Reply:
I'm fairly certain that this would work just fine in this case. However, since the serialization appears to be a bottleneck anyway, maybe we want another, more explicit mechanism for these types of objects, of which there are many. Something like explicit string templates to avoid any reflection, if this is actually the reason for the long processing time. This is not something that would be part of this pull-request however, so I guess this is fine for now.